File: rockpaper.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
char compdecision(){

    int randomInRange = 1 + (rand() % 3);
    
    if(randomInRange == 1){
        return 'r';
    }
    else if(randomInRange == 2){
        return 'p';
    }
    else{
        return 's';
    }
}
char userdecision(){
    char userinput;
    printf("Enter your choice: r-rock, p-paper, or s-scissors: ");
    scanf(" %c", &userinput);
    return userinput;
}
char compare(char user, char comp){
    
    if(user == comp){
        return 't';
    }
    else if(user == 'r' && comp == 's'){
        return 'w';
    }
    else if(user == 'p' && comp == 'r'){
        return 'w';
    }
    else if(user == 's' && comp == 'p'){
        return 'w';
    }
    else if(user == 'r' && comp == 'p'){
        return 'l';
    }
    else if(user == 'p' && comp == 's'){
        return 'l';
    }
    else if(user == 's' && comp == 'r'){
        return 'l';
    }
    else{
        return 'k';
    }
}
int main(){
        srand(time(0));
    char user, comp, result;
    int i = 0;
    int compscore = 0;
    int userscore = 0;
    printf("*****Welcome to Rock, Paper, Scissors!*****\n");
    do
    {
        /* code */

        user = userdecision();
         if (user != 'r' && user != 'p' && user != 's') {
            printf("Invalid input! Please enter 'r', 'p', or 's'.\n");
            continue; //  invalid input without incrementing `i`
        }
        comp = compdecision();
        result = compare(user, comp);
        printf("Computer chose: %c\n", comp);
        if(result == 't'){
            printf("It's a tie!\n");
        }
        else if(result == 'w'){
            printf("You win!\n");
            userscore++;
        }
        else if (result == 'l'){
            printf("You lose!\n");
            compscore++;
        }
        else{
            printf("Invalid input!\n");
        }
        i++;

    } while (i<3);
    printf("Final score - You: %d, Computer: %d\n", userscore, compscore);
    if (compscore > userscore)
    {
        /* code */
        printf("Computer wins the game!\n");
    }
    
    else if (compscore < userscore)
    {
        /* code */
        printf("You win the game!\n");
    }
    else
    {
        printf("It's a tie!\n");
    }
    
   
    return 0;
}

File: rockpaper.py
import random
def compdecision():
    import random
    comp = random.randint(1,3)
    if comp == 1:
        return "rock"
    elif comp == 2:
        return "paper"
    else:
        return "scissors"


def userdecision():
    user = input("Enter rock, paper, or scissors: ")
    return user
def gamedecision(user, comp):
    if user == comp:
        return "It's a tie!"
    elif (user == "rock" and comp == "scissors") or (user == "paper" and comp == "rock") or user == "scissors" and comp == "paper":
        return "You win!"
        
    else:
            return "computer win!"
    
    
def main():
    i=0
    compscore = 0
    userscore = 0
    
    while i<3:
        comp = compdecision()
        user = userdecision().lower()
        if user != "rock" and user != "paper" and user != "scissors":
            print("Invalid input! Try again.")
            continue
        print("Computer chose", comp)
        result = gamedecision(user, comp)
        print(result)
        if result == "computer win!":
            compscore += 1
        elif result == "You win!":
            userscore += 1
        i += 1

    print("\nGame Over!")
    print("Computer's score:", compscore)
    print("Your score:", userscore)    
    if compscore > userscore:
        print("Computer wins the game!")
    elif compscore < userscore:
        print("You win the game!")
    else:
        print("It's a tie!")

    

print("Welcome to Rock, Paper, Scissors!")
print("You will play 3 rounds against the computer.")
print("Enter 'rock', 'paper', or 'scissors' to make your choice.\n")
main() 
print("do you want to play again?/n yes or no")
play_again = input().lower()
while play_again == "yes":
    main() 
    print("do you want to play again?/n yes or no")
    play_again = input().lower()

print("Thanks for playing!")